enet: fsl_enet: Expose Timestamp of 1588_EVENT_IN

eggersn opened the pull request:
1588_EVENT_IN signals can latch the current timer value in TCCRn when a rising edge is detected. This pull request exposes the latched timer value via ENET_Ptp1588GetCapture.

stanislav-poboril commented on drivers/enet/fsl_enet.c:
```diff
@@ -3004,6 +3004,17 @@ void ENET_Ptp1588GetTimer(ENET_Type *base, enet_handle_t *handle, enet_ptp_time_
+ * @param base  ENET peripheral base address.
+ * @param channel The ENET PTP timer channel number.
+ */
+uint32_t ENET_Ptp1588GetCapture(ENET_Type *base, enet_ptp_timer_channel_t channel)
```
There is a function for reading of TCCR in fsl_enet.h:

```
/*!
 * @brief Sets the ENET PTP 1588 timer channel comparison value.
 *
 * @param base  ENET peripheral base address.
 * @param channel The PTP timer channel, see "enet_ptp_timer_channel_t".
 * @param cmpValue The compare value for the compare setting.
 */
static inline void ENET_Ptp1588SetChannelCmpValue(ENET_Type *base, enet_ptp_timer_channel_t channel, uint32_t cmpValue)
{
    base->CHANNEL[channel].TCCR = cmpValue;
}
```

I suggest to make the getter function more aligned with that:

- define in .h close to the above function, also as inline
- match the naming of the function and text in the comment with this one
- match formatting style (four spaces)

stanislav-poboril replied:
@mmahadevan108 I have never used this functionality, but looking at the manual it seems you either write compare value to the register in compare mode and read captured value in capture mode, so maybe just rename to ENET_Ptp1588GetChannelCapture or ENET_Ptp1588GetChannelCaptureValue to match with ENET_Ptp1588SetChannelCmpValue and place it beside it inlined in a header file. Thanks.

stanislav-poboril commented:
> Hello @stanislav-poboril, please help to review this PR, thanks.

Reviewed.
Note that current flow is that this is not going to be merged directly to GitHub, it will be integrated in the internal repository first and then the change will get here as part of the SDK release.

eggersn committed:
enet: fsl_enet: Expose Timestamp of 1588_EVENT_IN

1588_EVENT_IN signals can latch the current timer value in TCCRn when a
rising edge is detected. This commit exposes the latched timer value via
ENET_Ptp1588GetCapture.

Signed-off-by: Simon Egger <simon.egger@ipvs.uni-stuttgart.de>

ZhaoxiangJin commented:
> @ZhaoxiangJin Can you confirm this change has been pulled into the SDK repo?

Hello @stanislav-poboril, could you please help to pick this PR to internal SDK? so that we can merge the hal_nxp side PR.

stanislav-poboril commented:
> > @ZhaoxiangJin Can you confirm this change has been pulled into the SDK repo?
> 
> Hello @stanislav-poboril, could you please help to pick this patch to internal SDK? so that we can merge the hal_nxp side PR.

Hello, hopefully the internal SDK pull request will get merged tomorrow, I will let you know once it is done.

mmahadevan108 commented:
Can this be closed?

mcuxsusan commented:
Close pull request as the fix already released.
